packages/snaps-rpc-methods/src/permitted/requestPermissions.test.ts

```diff
@@ -0,0 +1,256 @@
+import { JsonRpcEngine } from '@metamask/json-rpc-engine';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  RequestPermissionsParams,
+  RequestPermissionsResult,
+} from '@metamask/snaps-sdk';
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+
+import { requestPermissionsHandler } from './requestPermissions';
+
+describe('requestPermissionsHandler', () => {
+  it('has the expected shape', () => {
+    expect(requestPermissionsHandler).toMatchObject({
+      methodNames: ['snap_requestPermissions'],
+      implementation: expect.any(Function),
+      hookNames: {
+        requestPermissions: true,
+      },
+    });
+  });
+
+  describe('implementation', () => {
+    it('returns the result received from the `requestPermissions` hook', async () => {
+      const { implementation } = requestPermissionsHandler;
+
+      const getOriginSnap = jest.fn().mockReturnValue({
+        manifest: {
+          dynamicPermissions: {
+            'endowment:network-access': {},
+          },
+        },
+      });
+      const requestPermissions = jest.fn().mockResolvedValue({ result: true });
```

Comment:
Let's test with a better mock that returns something closer to an actual return value